cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -731,16 +741,32 @@ impl LockedManifestCatalog {
             })
             .flatten()
             .filter_map(|resolved_pkg| {
-                let Some(descriptor) = manifest.pkg_descriptor_with_id(&resolved_pkg.install_id)
-                else {
-                    debug!(
-                        "Package {} is not in the manifest, skipping",
-                        resolved_pkg.install_id
-                    );
-                    return None;
-                };
-
-                Some(LockedPackageCatalog::from_parts(resolved_pkg, descriptor))
+                match manifest.pkg_descriptor_with_id(&resolved_pkg.install_id) {
```

Comment:
nonblocking: would it be simpler to use `catalog_pkg_descriptor_with_id` here? Most of these arms shouldn't be reachable

Although it might be even better to split up descriptors in `lock_manifest` such that `collect_package_groups` and `locked_packages_from_resolution` get slices of catalog descriptors rather than entire manifests. Is there any reason they need the whole manifest?

Reply:
When i worked on this i tried to keep changes to the existing locking minimal and address those with #1737.
Using `catalog_pkg_descriptor_with_id` probably makes sense either way indeed